File: VB-Examples/11_Formatting/ApplyGradientFillEffects.vb
Imports Spire.Xls
Imports System.ComponentModel
Imports System.Text
Imports Spire.Xls.Core

Namespace ApplyGradientFillEffects
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Set the Excel version to 2010
            workbook.Version = ExcelVersion.Version2010

            ' Get the first Worksheet from the Workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Get the CellRange object for cell B5
            Dim range As CellRange = sheet.Range("B5")

            ' Set the row height of the range to 50
            range.RowHeight = 50

            ' Set the column width of the range to 30
            range.ColumnWidth = 30

            ' Set the text in the range to "Hello"
            range.Text = "Hello"

            ' Set the horizontal alignment of the range to center
            range.Style.HorizontalAlignment = HorizontalAlignType.Center

            ' Set the fill pattern of the range to Gradient
            range.Style.Interior.FillPattern = ExcelPatternType.Gradient

            ' Set the fore color of the gradient to RGB(255, 255, 255)
            range.Style.Interior.Gradient.ForeColor = Color.FromArgb(255, 255, 255)

            ' Set the back color of the gradient to RGB(79, 129, 189)
            range.Style.Interior.Gradient.BackColor = Color.FromArgb(79, 129, 189)

            ' Apply a two-color horizontal gradient shading effect to the gradient
            range.Style.Interior.Gradient.TwoColorGradient(GradientStyleType.Horizontal, GradientVariantsType.ShadingVariants1)

            ' Specify the filename for the saved workbook
            Dim result As String = "ApplyGradientFillEffects_result.xlsx"

            ' Save the workbook to the specified file with Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer(result)
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/11_Formatting/ApplyGradientFillEffects.cs
using Spire.Xls;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Xls.Core;

namespace ApplyGradientFillEffects
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();
            workbook.Version = ExcelVersion.Version2010;

            //Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            //Get "B5" cell
            CellRange range =sheet.Range["B5"];

            //Set row height and column width
            range.RowHeight = 50;
            range.ColumnWidth = 30;
            range.Text = "Hello";

            //Set alignment style
            range.Style.HorizontalAlignment = HorizontalAlignType.Center;

            //Set gradient filling effects
            range.Style.Interior.FillPattern = ExcelPatternType.Gradient;
            range.Style.Interior.Gradient.ForeColor = Color.FromArgb(255, 255, 255);
            range.Style.Interior.Gradient.BackColor = Color.FromArgb(79, 129, 189);
            range.Style.Interior.Gradient.TwoColorGradient(GradientStyleType.Horizontal, GradientVariantsType.ShadingVariants1);

            // Save to
            String result = "ApplyGradientFillEffects_result.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the document
            FileViewer(result);
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}
